plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts

```diff
@@ -0,0 +1,173 @@
+		date: `${ month }-${ day }-${ year } ${ hours }:${ seconds }`,
+		catalog_visibility: ( document.querySelector(
+			'input[name="_visibility"]:checked'
+		) as HTMLInputElement )?.value,
+		featured: ( document.querySelector( '#_featured' ) as HTMLInputElement )
+			?.checked,
+	};
+};
+
+/**
+ * Prefix all object keys with a string.
+ *
+ * @param  obj    Object to create keys from.
+ * @param  prefix Prefix used before all keys.
+ * @return object
+ */
+const prefixObjectKeys = (
+	obj: { [ key: string ]: unknown },
+	prefix: string
+) => {
+	return Object.fromEntries(
+		Object.entries( obj ).map( ( [ k, v ] ) => [ `${ prefix }${ k }`, v ] )
+	);
+};
+
+/**
+ * Initialize all product screen tracks.
+ */
+export const initProductScreenTracks = () => {
+	const initialPublishingData = getPublishingWidgetData();
+
+	document
+		.querySelector( '#post-preview' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_preview_changes' );
+		} );
+
+	document
+		.querySelector( '.submitduplicate' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_copy', getProductData() );
+		} );
+
+	document
+		.querySelector( '.submitdelete' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_delete', getProductData() );
+		} );
+
+	document
+		.querySelectorAll(
+			'.edit-post-status, .edit-visibility, .edit-timestamp, .edit-catalog-visibility'
+		)
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', () => {
+				recordEvent( 'product_publish_widget_edit', {
+					...getPublishingWidgetData(),
+					...getProductData(),
+				} );
+			} );
+		} );
+
+	document.querySelector( '#publish' )?.addEventListener( 'click', () => {
+		recordEvent( 'product_publish_widget_save', {
+			...prefixObjectKeys( getPublishingWidgetData(), 'new_' ),
+			...prefixObjectKeys( initialPublishingData, 'current_' ),
+			...getProductData(),
+		} );
+	} );
+
+	document
+		.querySelectorAll( '.handle-order-lower, .handle-order-higher' )
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', ( event ) => {
+				const postBox = ( event.target as HTMLElement ).closest(
+					'.postbox'
+				);
+
+				if ( ! postBox ) {
+					return;
+				}
+
+				recordEvent( 'product_widget_order_change', {
+					widget: postBox.id,
+				} );
+			} );
+		} );
+
+	document
+		.querySelector( '#show-settings-link' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_screen_options_open' );
+		} );
+
+	document
+		.querySelectorAll( '#adv-settings input[type=checkbox]' )
+		.forEach( ( input ) => {
+			input.addEventListener( 'change', () => {
+				recordEvent( 'product_screen_elements', {
+					selected_element: ( input as HTMLInputElement ).value,
+					checkbox: ( input as HTMLInputElement ).checked,
+				} );
+			} );
+		} );
+
+	document
+		.querySelectorAll( 'input[name="screen_columns"]' )
+		.forEach( ( input ) => {
+			input.addEventListener( 'change', () => {
+				recordEvent( 'product_screen_elements', {
```

Comment:
I believe this should be `wcadmin_product_layout`?

Reply:
Good catch!